### PR TITLE
GEODE-8239 - Add gradle config to add 'Class-Path' and 'Dependent-Modules' attirbutes to manifest file

### PR DESCRIPTION
**Pull request by yozaner1324**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: gradle/publish-java.gradle [hunk 1]
```diff
@@ -28,3 +28,41 @@ publishing {
...
+          if (parentProject != null) {
+            def collect = parentProject.configurations.runtimeClasspath.collect { it.name - ".jar" }
+            runtimeList.removeAll(collect)
+            projectDependencies.removeAll(collect)
+          }
+        }
+      }
+
+      manifest {
+        attributes.put("Class-Path", runtimeList.join(' '))
```
Comment (rhoughton-pivot) [inline gradle/publish-java.gradle:61]: Is Attributes.put() what avoids re-writing the entire block?

File: gradle/publish-java.gradle [hunk 2]
```diff
@@ -28,3 +28,41 @@ publishing {
...
+              .collect { it.name - ".jar" }
+              .each { dependency ->
+                if (dependency.startsWith("geode-")) {
+                  projectDependencies.add(dependency)
+                } else {
+                  runtimeList.add(dependency)
+                }
+              }
+
+      projectDependencies.clone().each { projectDependency ->
```
Comment (rhoughton-pivot) [inline gradle/publish-java.gradle:48]: Can we get a comment block on the logic of what is being pruned out from this list? I get it, but I also consulted on it in the first place

File: gradle/publish-java.gradle (hunk 1 above)
Comment (yozaner1324) [reply gradle/publish-java.gradle:61]: Yes

Comment (kohlmu-pivotal) [reply gradle/publish-java.gradle:61]: yes..

File: gradle/publish-java.gradle (hunk 2 above)
Comment (kohlmu-pivotal) [reply gradle/publish-java.gradle:48]: ok.. we'll comment on what and how the pruning is happening.

Comment (rhoughton-pivot) [reply gradle/publish-java.gradle:55]: Perfect.

**Pull request merged**